### PR TITLE
Log files used to build config

### PR DESCRIPTION
**Pull request by 1nickt**
https://github.com/PerlDancer/Dancer2/issues/1593

### CODE REVIEW

**Comment by 1nickt** [comment]
@cromedome  This failure:
```
Unable to load class for Logger component Trap: Can't locate Dancer2/Logger/Trap.pm in @INC (you may need to install the Dancer2::Logger::Trap module) (@INC contains: /github/home/.cpanm/work/1615150101.14505/Dancer2-Plugin-Auth-Extensible-0.709/blib/lib /github/home/.cpanm/work/1615150101.14505/Dancer2-Plugin-Auth-Extensible-0.709/blib/arch /usr/local/lib/perl5/site_perl/5.32.0/x86_64-linux-gnu /usr/local/lib/perl5/site_perl/5.32.0 /usr/local/lib/perl5/vendor_perl/5.32.0/x86_64-linux-gnu /usr/local/lib/perl5/vendor_perl/5.32.0 /usr/local/lib/perl5/5.32.0/x86_64-linux-gnu /usr/local/lib/perl5/5.32.0 .) at /usr/local/lib/perl5/site_perl/5.32.0/Module/Runtime.pm line 314.
 at /usr/local/lib/perl5/site_perl/5.32.0/Dancer2/Core/App.pm line 215.
BEGIN failed--compilation aborted at t/no-login-handler.t line 17.
t/no-login-handler.t ................. 
```
is in a Auth::Extensible test. I first suspected that I was tripping an exception by calling `$self->log` indiscriminately, hence the second commit. The test still fails and I think the 2nd commit may be unneeded as I first thought.

**Comment by 1nickt** [comment]
@cromedome @bigpresh Fix for bug in A::E test that is exposed by this change : https://github.com/PerlDancer/Dancer2-Plugin-Auth-Extensible/pull/96

**Comment by cromedome** [comment]
:+1: Thanks Nick!

File: lib/Dancer2/Core/App.pm [hunk 1]
```diff
@@ -1111,6 +1111,11 @@ sub BUILD {
     $self->_init_hooks();
 }
 
+after BUILD => sub {
+    my $self = shift;
+    $self->log(core => 'Built config from files: ' . join(' ', @{$self->config_files}));
```
Comment (veryrusty) [inline lib/Dancer2/Core/App.pm]: Why have this inside an `after` wrapper, rather then inside the build method itself ?

Comment (1nickt) [reply lib/Dancer2/Core/App.pm]: My thought was to make sure that everything (config, logger) was built. But you're right, it's not necessary. Updating.

**Comment by 1nickt** [comment]
Opening a new PR with tests

**Pull request closed without merging**